### PR TITLE
send app insights trace level, use idsrv token validation in api hosting

### PR DESCRIPTION
**Pull request by alhardy**

### CODE REVIEW

File: sandbox/Sandbox.Api/appsettings.json [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
-      "Write": [ "write" ]
+      "Read": [ "read_sandbox" ],
+      "Write": [ "write_sandbox" ]
     }
   },
   "AppDomain": "SampleApp",
   "DataDogApiKey": "",
   "KeyVaultName": "",
-  "SharedAzureAppConfigurationEndpoint": ""
+  "SharedAzureAppConfigurationEndpoint": "https://staging-config-prospa-3p.azconfig.io"
```
Comment (luizbon) [inline sandbox/Sandbox.Api/appsettings.json:31]: Mistake?

File: src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs [hunk 2]
```diff
@@ -22,9 +25,17 @@ public static IServiceCollection AddProspaDefaultAuthenticationAndAuthorization(
...
+            var authOptions = new AuthOptions();
+            configuration.GetSection(nameof(AuthOptions)).Bind(authOptions);
+            services.AddAuthentication(IdentityServerAuthenticationDefaults.AuthenticationScheme)
+                    .AddIdentityServerAuthentication(
+                        options =>
+                        {
+                            options.Authority = authOptions.Authority;
+                            options.ApiName = authOptions.Audience;
+                            options.SupportedTokens = SupportedTokens.Both;
+                            options.JwtValidationClockSkew = TimeSpan.FromMinutes(5);
```
Comment (luizbon) [inline src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs:37]: Not a big deal, but this Clock Skew could be defined as part of the options

File: src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs [hunk 3]
```diff
@@ -22,9 +25,17 @@ public static IServiceCollection AddProspaDefaultAuthenticationAndAuthorization(
...
 
             // Add Authentication
-            var authenticationBuilder = services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme);
-            authenticationBuilder.Services.AddSingleton<IConfigureOptions<JwtBearerOptions>, JwtBearerOptionsSetup>();
-            authenticationBuilder.AddJwtBearer();
+            var authOptions = new AuthOptions();
+            configuration.GetSection(nameof(AuthOptions)).Bind(authOptions);
+            services.AddAuthentication(IdentityServerAuthenticationDefaults.AuthenticationScheme)
+                    .AddIdentityServerAuthentication(
+                        options =>
```
Comment (luizbon) [inline src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs:32]: It'll be better if we had an overload exposing this options, same as IDS does, this will give us more flexibility for edge cases.

File: sandbox/Sandbox.Api/appsettings.json (hunk 1 above)
Comment (alhardy) [reply sandbox/Sandbox.Api/appsettings.json]: > Mistake?

Removed test config endpoint, also removed authority to shared app config

File: src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs (hunk 2 above)
Comment (alhardy) [reply src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs:37]: > Not a big deal, but this Clock Skew could be defined as part of the options

Added here because It should be the same across the board. Allows us to managed through package updates

File: src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs (hunk 3 above)
Comment (alhardy) [reply src/Prospa.Extensions.AspNetCore.Hosting/StartupAuth.cs:32]: > It'll be better if we had an overload exposing this options, same as IDS does, this will give us more flexibility for edge cases.

Again this is why I was initially hesitant in such a package i..e baking in defaults. The question will always be what's client configurable for stock standard. For edge cases clients can just bring the code in. I think in all cases at the moment we use the same configuration. Let's review when the need arises
